parsl/app/futures.py

```diff
@@ -30,15 +38,23 @@ def parent_callback(self, parent_fu):
         Returns:
             - None
         """
-
         e = parent_fu._exception
         if e:
             self.set_exception(e)
         else:
             self.set_result(self.file_obj)
+            # only update the file object if it is a file
+            if self.data_flow_kernel.file_provenance and self.file_obj.scheme == 'file' and os.path.isfile(self.file_obj.filepath):
+                if not self.file_obj.timestamp:
```

Comment:
Does this stat mean that if there's a file transfer, we get the timestamp of the time at which the file was transferred rather than when it was created?